### ISSUE TITLE
Process hangs on manual request abortion

### ISSUE THREAD

**Issue by Vladislao** [tomas/needle#257]
It looks like inner timeout is not cleared when aborting request manually. Maybe I'm missing something? 

Here is code to reproduce:
```node
const needle = require("needle");
const log = require("why-is-node-running");

const readStream = needle
  .get("https://google.com", {
    read_timeout: 120000,
    follow_max: 10
  })
  .on("header", statusCode => {
    // do some checks, but for example purposes I'll just abort the request
    readStream.request.abort();
    readStream.destroy();
  });

setTimeout(() => {
  log();
}, 500);
```

**Comment by tomas** [comment]
You're right! We need to ensure all timers are cleared when the request is aborted

**Comment by tomas** [comment]
Done! Just pushed v2.2.3

**Comment by puzrin** [comment]
@tomas could you comment why `stream.request.abort();` not called automatically on `stream.destroy()`?

I got similar issue with mocha hang due missed timers cleanup. Just used `require('stream').pipeline(needle_stream, comsumer)` and called `consumer.destroy()` when no more data needed.

AFAIK, `pipeline()` is new recommended method to glue streams with guaranteed cleanup. It would be nice to support it.
